### PR TITLE
crit: fix bytes vs str issues with unix sock names

### PR DESCRIPTION
**Pull request by nviennot**
Improve the utf8 handling for unix socket

### CODE REVIEW

**Comment by adrianreber** [comment]
Is there a certain image that needs this fix. Would be nice to have it included in our CI. Especially with the parallel Python 2 and Python 3 support it would be good to have a test for this change.

At https://github.com/checkpoint-restore/criu/blob/criu-dev/scripts/ci/run-ci-tests.sh#L270 is a good place to add additional images to the recode test which triggered a lot of the other recent changes to the Python code.

File: lib/py/images/pb2dict.py [hunk 1]
```diff
@@ -261,11 +261,11 @@ def decode_base64(value):
 
 
 def encode_unix(value):
-    return quopri.encodestring(value)
+    return quopri.encodestring(value).decode('utf8')
 
 
 def decode_unix(value):
-    return quopri.decodestring(value)
+    return quopri.decodestring(value.encode('utf'))
```
Comment (Snorch) [inline lib/py/images/pb2dict.py:268]: Probably: s/utf/utf8/

**Review by Snorch — COMMENTED** [review]
Hi, @nviennot 

Your commit message lacks details:

```
 crit: fix bytes vs str issues with unix sock names

Improve the utf8 handling for unix socket
```

You can put information about which exact errors you get into commit message, this would help people which would face same issues with older criu version to understand that they need your patch. Also it would help review the patch.

Also it would be nice to have a description of how you fix the problem, so that we can understand that what you want to do corresponds to what you actually do.

Please take a look on https://github.com/checkpoint-restore/criu/blob/criu-dev/CONTRIBUTING.md#describe-your-changes our contribution guide.

**Comment by nviennot** [comment]
You are right, I have to revisit this. We probably would benefit from test cases as well.

**Comment by adrianreber** [comment]
@nviennot Do you want to rebase this?

**Pull request closed without merging**